src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TypeFixupSignature.cs

```diff
@@ -39,11 +42,114 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                 EcmaModule targetModule = factory.SignatureContext.GetTargetModule(_typeDesc);
                 SignatureContext innerContext = dataBuilder.EmitFixup(factory, _fixupKind, targetModule, factory.SignatureContext);
                 dataBuilder.EmitTypeSignature(_typeDesc, innerContext);
+
+                if (_fixupKind == ReadyToRunFixupKind.Check_TypeLayout)
+                {
+                    EncodeTypeLayout(dataBuilder, _typeDesc);
+                }
             }
 
             return dataBuilder.ToObjectData();
         }
 
+        private static void EncodeTypeLayout(ObjectDataSignatureBuilder dataBuilder, TypeDesc type)
+        {
+            Debug.Assert(type.IsValueType);
+            MetadataType defType = (MetadataType)type;
+
+            int pointerSize = type.Context.Target.PointerSize;
+            int size = defType.InstanceFieldSize.AsInt;
+            int alignment = GetClassAlignmentRequirement(defType);
+            ReadyToRunTypeLayoutFlags flags = ReadyToRunTypeLayoutFlags.READYTORUN_LAYOUT_Alignment | ReadyToRunTypeLayoutFlags.READYTORUN_LAYOUT_GCLayout;
+            if (alignment == pointerSize)
+            {
+                flags |= ReadyToRunTypeLayoutFlags.READYTORUN_LAYOUT_Alignment_Native;
+            }
+
+            if (!defType.ContainsGCPointers)
+            {
+                flags |= ReadyToRunTypeLayoutFlags.READYTORUN_LAYOUT_GCLayout_Empty;
+            }
+
+            bool isHfa = defType.IsHfa && defType.Context.Target.Architecture == TargetArchitecture.ARM || defType.Context.Target.Architecture == TargetArchitecture.ARM64;
```

Comment:
I may be missing something but shouldn't we parenthesize the two architecture checks - or is the <code>defType.IsHfa</code> check only relevant for 32-bit ARM?

Reply:
Good catch - I moved this code and it doesn't need parentheses in the new location